WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewModel.swift

```diff
@@ -0,0 +1,134 @@
+
+    private let presentNoticeSubject: PassthroughSubject<SimplePaymentsNotice, Never> = PassthroughSubject()
+    private let analytics: Analytics = ServiceLocator.analytics
+
+    private var subscriptions = Set<AnyCancellable>()
+
+    private var siteID: Int64 {
+        stores.sessionManager.defaultStoreID ?? 0
+    }
+
+    init(didChangeShouldShow: ((CardReaderSettingsTriState) -> Void)?,
+         connectionAnalyticsTracker: CardReaderConnectionAnalyticsTracker,
+         stores: StoresManager = ServiceLocator.stores) {
+        self.didChangeShouldShow = didChangeShouldShow
+        self.connectionAnalyticsTracker = connectionAnalyticsTracker
+        self.stores = stores
+
+        beginConnectedReaderObservation()
+    }
+
+    /// Set up to observe readers connecting / disconnecting
+    ///
+    private func beginConnectedReaderObservation() {
+        // This completion should be called repeatedly as the list of connected readers changes
+        let connectedAction = CardPresentPaymentAction.observeConnectedReaders() { [weak self] readers in
+            guard let self = self else {
+                return
+            }
+            self.connectedReader = readers.isNotEmpty ? .isTrue : .isFalse
+            self.reevaluateShouldShow()
+        }
+        stores.dispatch(connectedAction)
+    }
+
+    private func startTestPayment() {
+        loading = true
+        let action = OrderAction.createSimplePaymentsOrder(siteID: siteID,
+                                                           status: .pending,
+                                                           amount: Constants.tapToPayTryAPaymentAmount,
+                                                           taxable: false) { [weak self] result in
+            guard let self = self else { return }
+            self.loading = false
+
+            switch result {
+            case .success(let order):
+                self.summaryViewModel = TryAPaymentSummaryViewModel(
+                    simplePaymentSummaryViewModel: SimplePaymentsSummaryViewModel(order: order,
+                                                                                  providedAmount: order.total,
+                                                                                  presentNoticeSubject: self.presentNoticeSubject),
+                    siteID: self.siteID,
+                    orderID: order.orderID)
+
+
+            case .failure(let error):
+                self.presentNoticeSubject.send(.error(Localization.errorCreatingTestPayment))
+                self.analytics.track(event: WooAnalyticsEvent.PaymentsFlow.paymentsFlowFailed(flow: .tapToPayTryAPayment,
+                                                                                              source: .tapToPayTryAPaymentPrompt))
+                DDLogError("⛔️ Error creating Tap to Pay try a payment order: \(error)")
+            }
+        }
+        ServiceLocator.stores.dispatch(action)
```

Comment:
We don't need to access to ServiceLocator here, right?
```suggestion
        stores.dispatch(action)
```

Reply:
Thanks, good catch. Fixed in c1169cb